solidity/contracts/vortex/VortexBurner.sol

```diff
@@ -0,0 +1,421 @@
+    }
+
+    /**
+     * @dev ETH receive callback
+     */
+    receive() external payable {}
+
+    /**
+     * @dev returns the burn incentive fee and its maximum amount
+     *
+     * @return the burn incentive fee and its maximum amount
+     */
+    function burnIncentiveFee() external view returns (uint32, uint256) {
+        return (_burnIncentiveFee, _maxBurnIncentiveFeeAmount);
+    }
+
+    /**
+     * @dev allows the owner to set the burn incentive fee and its maximum amount
+     *
+     * @param newBurnIncentiveNetworkFee the percentage of the converted network tokens to be sent to the caller of the burning event (in units of PPM)
+     * @param newMaxBurnIncentiveFeeAmount the maximum incentive fee to be sent to the caller of the burning event
+     */
+    function setBurnIncentiveFee(uint32 newBurnIncentiveNetworkFee, uint256 newMaxBurnIncentiveFeeAmount)
+        external
+        ownerOnly
+        validFee(newBurnIncentiveNetworkFee)
+    {
+        emit BurnIncentiveFeeUpdated(
+            _burnIncentiveFee,
+            newBurnIncentiveNetworkFee,
+            _maxBurnIncentiveFeeAmount,
+            newMaxBurnIncentiveFeeAmount
+        );
+
+        _burnIncentiveFee = newBurnIncentiveNetworkFee;
+        _maxBurnIncentiveFeeAmount = newMaxBurnIncentiveFeeAmount;
+    }
+
+    /**
+     * @dev returns the total amount of the burned governance tokens
+     *
+     * @return total amount of the burned governance tokens
+     */
+    function totalBurntAmount() external view returns (uint256) {
+        return _totalBurntAmount;
+    }
+
+    /**
+     * @dev converts the provided tokens to governance tokens and burns them
+     *
+     * @param tokens the tokens to convert
+     */
+    function burn(IERC20[] calldata tokens) external protected {
+        ITokenHolder feeWallet = networkFeeWallet();
+
+        // retrieve the burning strategy
+        Strategy memory strategy = burnStrategy(tokens, feeWallet);
+
+        // withdraw all token/ETH amounts to the contract
+        feeWallet.withdrawTokensMultiple(tokens, address(this), strategy.amounts);
+
+        // convert all amounts to the network token and record conversion amounts
+        IBancorNetwork network = bancorNetwork();
+
+        for (uint256 i = 0; i < strategy.paths.length; ++i) {
+            // avoid empty conversions
+            uint256 amount = strategy.amounts[i];
+            if (amount == 0) {
+                continue;
+            }
+
+            address[] memory path = strategy.paths[i];
+            IERC20 token = IERC20(path[0]);
+            uint256 value = 0;
+
+            if (token == _networkToken || token == _govToken) {
+                // if the source token is the network or the governance token, we won't try to convert it, but rather
+                // include its amount in either the total amount of tokens to convert or burn.
+                continue;
+            }
+
+            if (token != NATIVE_TOKEN_ADDRESS) {
```

Comment:
maybe flip the condition/blocks here so that we'll use equality and not inequality?

Reply:
👍